src/promise/Promise.php

```diff
@@ -23,10 +23,11 @@
 
 namespace pocketmine\promise;
 
+use function count;
 use function spl_object_id;
 
 /**
- * @phpstan-template TValue
+ * @phpstan-template-covariant TValue
```

Comment:
are you sure a covariant `TValue` really makes sense? it seems that `TValue` would be contravariant in `onCompletion`.

Reply:
Without, all analystic tools will raised some error when we pass null on mixed attended type

Reply:
I'd also prefer if it could be avoided. It's confusing and I feel like it's probably not necessary now that we can do `Promise<*>` in newer versions of PHPStan.